### PR TITLE
Remove git fetch with destination refspec feature

### PR DESCRIPTION
**Pull request by b10n1k**
As per git documentation the refspec can formed as <src>:<dst> but dst cannot
use arbitrary expressions (SHA) but an actual ref must be named. Its usage
in the `_git_clone` mistakely adds a SHA ref as destination creating a branch
with its name, which is not supported by git.
   
As far as now the feature is not used in OpenQA so it can be safely removed.

https://progress.opensuse.org/issues/164937

### CODE REVIEW

**Review by perlpunk — COMMENTED** [review]
We talked about it earlier today and decided that we don't need the local branches at all.

File: lib/OpenQA/Task/Git/Clone.pm [hunk 1]
```diff
@@ -119,7 +119,8 @@ sub _git_clone ($job, $ctx, $path, $url) {
     }
     else {
         # updating local branch to latest remote branch version
-        _git_fetch($path, "$requested_branch:$requested_branch");
+        my $dst = 'clone_' . substr($requested_branch, 0, 7);
```
Comment (perlpunk) [inline lib/OpenQA/Task/Git/Clone.pm:122]: This could result in duplicates if branch names have common prefixes

**Review by perlpunk — CHANGES_REQUESTED** [review]
See comments

File: lib/OpenQA/Task/Git/Clone.pm [hunk 2]
```diff
@@ -119,7 +119,8 @@ sub _git_clone ($job, $ctx, $path, $url) {
...
+        _git_fetch($path, "$requested_branch:$dst");
```
Comment (perlpunk) [inline lib/OpenQA/Task/Git/Clone.pm:123]: This should do it:
```suggestion
        _git_fetch($path, $requested_branch);
```

Comment (perlpunk) [reply lib/OpenQA/Task/Git/Clone.pm]: Why did you ignore this suggestion?

File: t/14-grutasks.t [hunk 3]
```diff
@@ -671,7 +671,6 @@ subtest 'git clone' => sub {
     is $res->{result}, 'Job successfully executed', 'minion job result indicates success';
     like $mocked_git_calls[3], qr'git -C /etc/ fetch origin master', 'fetch origin master for /etc/';
     like $mocked_git_calls[4], qr'reset --hard origin/master', 'reset origin/master for /etc/';
-    like $mocked_git_calls[8], qr'git -C /root/ fetch origin foobranch:foobranch', 'fetch non-default branch';
```
Comment (perlpunk) [inline t/14-grutasks.t:674]: Why did you remove this test? Don't we still want to test that command? Just that it changed?

Comment (b10n1k) [reply t/14-grutasks.t:674]: We do not use any more. there is not such call with refspec in this format. Unless you want to test that it fails!

Comment (perlpunk) [reply t/14-grutasks.t:674]: The call is still there. I can see it if I checkout your PR.
It's just not `git -C /root/ fetch origin foobranch:foobranch` anymore, but `git -C /root/ fetch origin foobranch`.

File: lib/OpenQA/Task/Git/Clone.pm (hunk 2 above)
Comment (perlpunk) [reply lib/OpenQA/Task/Git/Clone.pm]: Ok, I think your change is actually the same as my suggestion, just that you shortened it because the fetch is now the same call for both cases.

File: t/14-grutasks.t (hunk 3 above)
Comment (b10n1k) [reply t/14-grutasks.t:674]: bring it back on f8e3adb2b. I had remove it as I dont think is much different from the test 2 lines above.

Comment (perlpunk) [reply t/14-grutasks.t:674]: Sure, but it's a different use case.
And as we saw in [the other pull request](https://github.com/os-autoinst/openQA/pull/5863#discussion_r1725013852), we actually would have produced an error in production because we didn't test all elements in that array.
Removing an existing test is rarely a good idea.

**Review by perlpunk — COMMENTED** [review]
Will review again when conflicts are resolved

**Review by perlpunk — APPROVED** [review]

**Comment by perlpunk** [comment]
Added not-ready label to avoid having trouble reverting #5880

**Comment by perlpunk** [comment]
Sorry, we had to revert #5880 so I guess you need to resolve conflicts again...

**Comment by Martchus** [comment]
Or wait until we can reinstate the refactoring. Otherwise it'll be a back and forth which is probably not worth it.

**Comment by perlpunk** [comment]
The refactoring will include some more changes which will cause more conflicts with this PR, so conflicts will appear in both cases possiibly

**Review by perlpunk — APPROVED** [review]
